text/0000-deprecation-ember-string.md

```diff
@@ -0,0 +1,62 @@
+- Start Date: 2017-07-14
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This PR proposes the deprecation of `Ember.String`. The following methods will be moved to an addon:
+
+- `camelize`
+- `capitalize`
+- `classify`
+- `dasherize`
+- `decamelize`
+- `fmt`
+- `loc`
+- `underscore`
+- `w`
+
+The remaining methods (`htmlSafe` and `isHTMLSafe`) will be moved to `@ember/component`.
+
+In both cases, the String prototype won't be extended anymore.
```

Comment:
@simonihmig You bring up a good point, but just to bring a different perspective, I don't think the point of a codmod is really to eliminate all human review of code change, but rather to remove the tedious step of changing everything yourself.

Use it, commit where you are sure the mod is correct, and review places you are unsure about.

I'm with @joukevandermaas on this - I rather have this not extend string by default, but leave it in as a feature flag that will throw deprecations from the string prototype methods (wrap methods in a deprecation when merging onto prototype). Right now there is a feature flag for extending prototypes in general, but maybe this RFC breaks string prototype into it's own feature flag. I think that makes sense, because in my mind, each group of prototype extensions exist for a very different reason. I see String extensions as utility, where as function extensions are syntactical sugar, and array functions were a mix of syntactical sugar and polyfilling new Array methods.